test/e2e/test_storage.go

```diff
@@ -0,0 +1,181 @@
+)
+
+// makeBucket will get an existed bucket or create a new one if it is not existed
+func makeBucket(t *testing.T) string {
+	ctx := context.Background()
+	project := os.Getenv(ProwProjectKey)
+	bucketPrefix := "storage-e2e-test"
+	if project == "" {
+		t.Fatalf("failed to find %q in envvars", ProwProjectKey)
+	}
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to create storage client, %s", err.Error())
+	}
+	it := client.Buckets(ctx, project)
+	bucketName := helpers.AppendRandomString(bucketPrefix)
+	for {
+		bucketAttrs, err := it.Next()
+		if err == iterator.Done {
+			// Create a new bucket if there is no qualified existing bucket
+			bucket := client.Bucket(bucketName)
+			if e := bucket.Create(ctx, project, &storage.BucketAttrs{}); e != nil {
+				t.Fatalf("failed to create bucket, %s", e.Error())
+			}
+			break
+		}
+		if err != nil {
+			t.Fatalf("failed to list buckets, %s", err.Error())
+		}
+		if strings.HasPrefix(bucketAttrs.Name, bucketPrefix) {
+			bucketName = bucketAttrs.Name
+			break
+		}
+	}
+	return bucketName
+}
+
+func getBucketHandler(t *testing.T, bucketName string) *storage.BucketHandle {
+	ctx := context.Background()
+	// Prow sticks the project in this key
+	project := os.Getenv(ProwProjectKey)
+	if project == "" {
+		t.Fatalf("failed to find %q in envvars", ProwProjectKey)
+	}
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to create pubsub client, %s", err.Error())
+	}
+	return client.Bucket(bucketName)
+}
+
+func StorageWithTestImpl(t *testing.T, packages map[string]string) {
+	ctx := context.Background()
+
+	bucketName := makeBucket(t)
+	storageName := bucketName + "-storage"
+	targetName := bucketName + "-target"
+
+	client := Setup(t, true)
+	defer TearDown(client)
+
+	config := map[string]string{
+		"namespace":  client.Namespace,
+		"storage":    storageName,
+		"bucket":     bucketName,
+		"targetName": targetName,
+		"targetUID":  uuid.New().String(),
+	}
+	for k, v := range packages {
+		config[k] = v
+	}
+
+	installer := NewInstaller(client.Dynamic, config,
+		EndToEndConfigYaml([]string{"storage_test", "istio"})...)
+
+	// Create the resources for the test.
+	if err := installer.Do("create"); err != nil {
+		t.Errorf("failed to create, %s", err)
+		return
+	}
+
+	//Delete deferred.
+	defer func() {
+		if err := installer.Do("delete"); err != nil {
+			t.Errorf("failed to delete, %s", err)
+		}
+		// Just chill for tick.
+		time.Sleep(10 * time.Second)
+	}()
+
+	gvr := schema.GroupVersionResource{
+		Group:    "events.cloud.run",
+		Version:  "v1alpha1",
+		Resource: "storages",
+	}
+
+	jobGVR := schema.GroupVersionResource{
+		Group:    "batch",
+		Version:  "v1",
+		Resource: "jobs",
+	}
+
+	if err := client.WaitForResourceReady(client.Namespace, storageName, gvr); err != nil {
+		t.Error(err)
+	}
+
+	// Add a file in the bucket
+	bucketHandler := getBucketHandler(t, bucketName)
+	wc := bucketHandler.Object("testFile").NewWriter(ctx)
```

Comment:
I think the test should be you generate a random file name, and pass this as the knock down key on the target. Otherwise you don't know the target it falling down because of _this_ file upload or something else.